src/components/client/core/Flyout/Flyout.tsx

```diff
@@ -0,0 +1,83 @@
+import { Portal } from "@ark-ui/react";
+import { useState } from "react";
+import { FiX as CloseIcon } from "react-icons/fi";
+
+import Button from "components/client/core/Button/Button";
+import Icon from "components/client/core/Icon/Icon";
+import {
+  Flyout as PrimitiveFlyout,
+  FlyoutArrow,
+  FlyoutArrowTip,
+  FlyoutCloseTrigger,
+  FlyoutContent,
+  FlyoutDescription,
+  FlyoutPositioner,
+  FlyoutTitle,
+  FlyoutTrigger,
+} from "components/primitives";
+import { flyout } from "generated/panda/recipes";
+import { useIsMounted } from "lib/hooks";
+
+import type { FlyoutProps } from "components/primitives";
+import type { ReactNode } from "react";
+
+export interface Props extends FlyoutProps {
+  trigger: ReactNode;
+  title?: string;
+  children: ReactNode;
+}
+
+const Flyout = ({ trigger, title, children, ...rest }: Props) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  const classNames = flyout();
+
+  const isMounted = useIsMounted();
+
+  if (!isMounted) return null;
+
+  return (
+    <PrimitiveFlyout
+      open={isOpen}
+      onClose={() => setIsOpen(false)}
+      portalled
+      {...rest}
+    >
+      <FlyoutTrigger asChild onClick={() => setIsOpen(!isOpen)}>
+        {trigger}
+      </FlyoutTrigger>
+
+      <Portal>
+        <FlyoutPositioner className={classNames.positioner}>
+          <FlyoutContent className={classNames.content}>
+            <FlyoutArrow className={classNames.arrow}>
+              <FlyoutArrowTip className={classNames.arrowTip} />
+            </FlyoutArrow>
+            {title && (
+              <FlyoutTitle className={classNames.title}>{title}</FlyoutTitle>
+            )}
+            <FlyoutDescription className={classNames.description}>
+              {children}
+            </FlyoutDescription>
+
+            <FlyoutCloseTrigger asChild onClick={() => setIsOpen(false)}>
+              <Button
+                pos="absolute"
+                top={2}
+                right={2}
+                bgColor={{
+                  base: "inherit",
+                  _hover: "bg.emphasized",
+                }}
+              >
+                <Icon as={CloseIcon} color="fg.default" />
+              </Button>
+            </FlyoutCloseTrigger>
```

Comment:
One thing I think we could do to simplify this close trigger pattern is to eliminate the need for `asChild` prop. Recently did this with the `Toast` component and I think it cleans it up a bit more. What I did was:

1) Add the button style props into the defined recipe (in this case `Flyout.recipe.ts` has `closeTrigger` already).
2) Simplify component structure like so:

```ts
<FlyoutCloseTrigger className={classNames.closeTrigger} onClick={() => setIsOpen(false)}>
    <Icon as={CloseIcon} color="fg.default" />
</FlyoutCloseTrigger>
```

This is merely a thought though. Let me know what you think!

Reply:
I think that is way cleaner. Let me peek at your toast branch quickly.

Reply:
Nice! Yeah the big thing is to just remember to add the `borderRadius` and `cursor` styles in as well since they are not baked in without our custom `Button` there anymore. Let me know!